feat: STT-ChatGPT-TTS 엔드포인트 연결

jayn2u opened the pull request:
- FastAPI 서버에서 만든 `STT-ChatGPT-TTS` 엔드포인트가 있습니다.
- 다른 API는 테스트 및 실험용으로 만들었는데, 해당 API는 서비스에서 실질적으로 사용되는 로직이 되겠습니다.
- 클라이언트에서 음성 녹음을 받은 데이터를 ChatGPT를 통해 응답을 받고, 받은 응답을 다시 음성 파일로 만들어서 클라이언트에게 주어야 합니다.
- 로컬에서 만든 API를 토큰 없이 어떻게 확인하는지 모르겠네요...?

jayn2u committed:
feat: STT-ChatGPT-TTS MP3 파일 처리 기능 추가

STT로 변환한 텍스트를 기반으로 ChatGPT 응답을 생성하고 이를 MP3 파일로 반환하는 기능을 추가했습니다. 관련 프로퍼티 및 새로운 API 엔드포인트가 구현되었습니다.

jayn2u committed:
refactor: import 정리 및 코드 스타일 개선

불필요한 순환 및 정렬되지 않은 import들을 정리하고, 코드 가독성을 위해 줄바꿈과 포매팅을 수정했습니다. 이에 따라 코드 유지보수성과 일관성이 향상되었습니다.

jayn2u committed:
feat: 오류 응답 정의 및 API 응답 명세 개선

LlmErrorResponse 클래스를 추가하여 오류 응답 모델을 정의하였고, LlmController의 API 응답 명세에 인증 실패와 관련된 응답 정보를 상세히 명시했습니다. 이를 통해 API 문서화 및 클라이언트와의 명확한 통신이 가능해졌습니다.

jayn2u committed:
feat: STT-ChatGPT-TTS 기능 추가

사용자가 업로드한 오디오 파일로 STT, ChatGPT, TTS를 연결하여 응답을 처리하는 기능을 추가했습니다. 새로운 탭 및 UI 섹션을 추가하고 관련 이벤트 핸들러를 구현했습니다.

jayn2u commented:
# 업데이트

- `force push`로 업데이트 사항 반영했습니다.
- 혹시 강제보다 더 안정적인 방법으로 맞추는 방법을 아시나요?
- 사전에 푸시한 브랜치와 로컬에서 새롭게 `rebase`된 사항을 푸시 올리는 것을 어떻게 안정적으로 할 수 있죠?

kwonyj1022 commented on src/main/java/com/newworld/saegil/authentication/controller/AuthenticationController.java:
```diff
@@ -119,7 +122,8 @@ public ResponseEntity<ValidateTokenResponse> validateToken(
     ) {
-        final TokenRefreshResult result = authenticationService.refreshToken(LocalDateTime.now(), request.refreshToken());
+        final TokenRefreshResult result = authenticationService.refreshToken(LocalDateTime.now(),
+                request.refreshToken());
```
### A
이 개행은 뭐죠? 안 하는게 더 가독성이 좋은 것 같아요.
그리고 혹시 인텔리제이에서 커밋 전에 자동으로 자동정렬 기능 수행하도록 해놨나요? 저는 이렇게 pr 범위가 아닌 사항에 대해 수정이 이루어지거나, 저 위에 import 순서가 뒤죽박죽 되는 것, ide가 맘대로 띄어쓰기 하는 상황을 방지하기 위해 꺼 놓는 편입니다.
지금 file changes가 63개인데 이 중에 유효한 file changes는 몇 개 없는 것 같아요. 특히 import나 개행은 각자의 ide 설정값에 따라 코드정렬할 때마다 달라서 모든 파일을 자동정렬하는 기능은 더더욱 안하는 편이기도 하구요.

kwonyj1022 commented on src/main/java/com/newworld/saegil/configuration/PropertiesConfiguration.java:
```diff
@@ -1,12 +1,14 @@
 @Configuration
-@EnableConfigurationProperties({JwtProperties.class, KakaoOAuth2Properties.class, ProxyProperties.class, CorsProperties.class})
+@EnableConfigurationProperties({JwtProperties.class, KakaoOAuth2Properties.class, ProxyProperties.class,
+        CorsProperties.class})
```
### A
이것도 아예 한 줄에 다 적거나 개행을 할 거면 한 개씩 다 개행을 해주면 좋을 것 같아요.
```suggestion
@EnableConfigurationProperties({
        JwtProperties.class, 
        KakaoOAuth2Properties.class, 
        ProxyProperties.class, 
        CorsProperties.class
})
```

kwonyj1022 commented on src/main/java/com/newworld/saegil/exception/GlobalExceptionHandler.java:
```diff
@@ -26,7 +27,8 @@ public ResponseEntity<ExceptionResponse> handleExceptionInternal(
         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
-                             .body(new ExceptionResponse(ex.getMessage() == null ? "Internal Server Error" : ex.getMessage()));
+                             .body(new ExceptionResponse(
+                                     ex.getMessage() == null ? "Internal Server Error" : ex.getMessage()));
```
### A
이것도 그냥 한 줄에 적거나 저 message 부분을 변수로 뺀 다음 사용하거나 아래처럼 하는게 더 좋아 보여요
```suggestion
                             .body(new ExceptionResponse(
                                     ex.getMessage() == null ? "Internal Server Error" : ex.getMessage()
                             ));
```

kwonyj1022 commented on src/main/java/com/newworld/saegil/llm/controller/LlmController.java:
```diff
@@ -125,4 +130,43 @@ public ResponseEntity<ChatGptResponse> chatGptFromFile(@RequestPart("file") Mult
+            produces = MediaType.APPLICATION_OCTET_STREAM_VALUE
+    )
+    public ResponseEntity<Resource> speechChatGptResponseFromFile(
+            @RequestPart("file") MultipartFile multipartFile) {
```
### A
```suggestion
    public ResponseEntity<Resource> speechChatGptResponseFromFile(
            @RequestPart("file") MultipartFile multipartFile
    ) {
```

kwonyj1022 commented on src/main/java/com/newworld/saegil/llm/controller/LlmErrorResponse.java:
```diff
@@ -0,0 +1,14 @@
+@Getter
+@AllArgsConstructor
+@Schema(description = "오류 응답")
+public class LlmErrorResponse {
```
### R
dto는 record 타입을 사용하고 있습니다!

### C
그나저나 이 클래스는 왜 필요한거죠? 에러 응답 관련해서는 예외를 터뜨린 다음 GlobalExceptionHandler에서 핸들링하고 있어요. 에러 응답은 모두 ExceptionResponse 클래스를 공통적으로 사용하고 있어요.

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/domain/Notice.java:
```diff
@@ -46,7 +46,8 @@ public class Notice {
 
-    public Notice(final String title, final String content, final NoticeType noticeType, final LocalDate date, final String webLink) {
+    public Notice(final String title, final String content, final NoticeType noticeType, final LocalDate date,
+            final String webLink) {
```
### A
```suggestion
    public Notice(
            final String title, 
            final String content, 
            final NoticeType noticeType, 
            final LocalDate date,
            final String webLink
    ) {
```

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/domain/crawler/HanaGeneralCrawler.java:
```diff
@@ -68,7 +70,8 @@ public List<Notice> crawl(final NoticeType noticeType, final LocalDate lastDate)
         );
-        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(), crawlEndTime - crawlStartTime);
+        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(),
+                crawlEndTime - crawlStartTime);
```
### A
여기도 개행 안 하는게 좋아보여요

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/domain/crawler/HanaPressReleaseCrawler.java:
```diff
@@ -63,7 +65,8 @@ public List<Notice> crawl(final NoticeType noticeType, final LocalDate lastDate)
         );
-        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(), crawlEndTime - crawlStartTime);
+        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(),
+                crawlEndTime - crawlStartTime);
```
### A
여기도!

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/domain/crawler/HanaTenderNoticeCrawler.java:
```diff
@@ -62,7 +64,8 @@ public List<Notice> crawl(final NoticeType noticeType, final LocalDate lastDate)
         );
-        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(), crawlEndTime - crawlStartTime);
+        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(),
+                crawlEndTime - crawlStartTime);
```
### A
여기도!

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/domain/crawler/UniKoreaCrawler.java:
```diff
@@ -62,7 +64,8 @@ public List<Notice> crawl(final NoticeType noticeType, final LocalDate lastDate)
         );
-        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(), crawlEndTime - crawlStartTime);
+        log.info("{} {} 크롤링 소요 시간: {}ms", noticeType.getSource(), noticeType.getCategory(),
+                crawlEndTime - crawlStartTime);
```
### A
여기도!

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/service/NoticeService.java:
```diff
@@ -58,9 +60,11 @@ private List<Notice> crawlAndExtractNewNotices(final NoticeCrawler crawler, fina
+                                                                                        .noneMatch(
+                                                                                                latestNotice -> latestNotice.hasSameTitle(
+                                                                                                        newNotice))
+                                                      ).toList();
```
### A
```suggestion
        final List<Notice> newNotices =
                crawledNotices.stream()
                              .filter(newNotice ->
                                      latestNotices.stream()
                                                   .noneMatch(latestNotice -> latestNotice.hasSameTitle(newNotice))
                              ).toList();
```

kwonyj1022 commented on src/main/resources/application-local.yml:
```diff
@@ -48,7 +48,8 @@ proxy:
-#    chatgpt-from-stt-text-path: /chatgpt/stt
-#    chatgpt-from-audio-url-path: /chatgpt/audio
+    #    chatgpt-from-stt-text-path: /chatgpt/stt
+    #    chatgpt-from-audio-url-path: /chatgpt/audio
```
### A
필요없어진 거니 지워버려도 될 것 같아요

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/domain/OAuth2HandlerCompositeTest.java:
```diff
@@ -1,16 +1,16 @@
 package com.newworld.saegil.authentication.domain;
 
-import com.newworld.saegil.security.oauth2.KakaoOAuth2Handler;
+import static org.assertj.core.api.Assertions.*;
```
### C
import에 와일드카드 쓰는 건 안 좋다고 생각합니다.

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/domain/OAuth2TypeTest.java:
```diff
@@ -1,14 +1,14 @@
 package com.newworld.saegil.authentication.domain;
 
+import static org.assertj.core.api.Assertions.*;
```
### C
여기도요!

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -1,5 +1,23 @@
 
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.BDDMockito.*;
```
### C
여기도요!

kwonyj1022 commented on src/test/java/com/newworld/saegil/security/jwt/JwtTokenProcessorTest.java:
```diff
@@ -1,21 +1,22 @@
-import com.newworld.saegil.authentication.domain.InvalidTokenException;
-import com.newworld.saegil.authentication.domain.TokenType;
-import io.jsonwebtoken.Claims;
+import static org.assertj.core.api.Assertions.*;
```
### C
여기도요!

kwonyj1022 commented on src/test/java/com/newworld/saegil/user/domain/UserTest.java:
```diff
@@ -1,14 +1,14 @@
 package com.newworld.saegil.user.domain;
 
-import com.newworld.saegil.authentication.domain.OAuth2Type;
+import static org.assertj.core.api.Assertions.*;
```
### C
여기도요!

kwonyj1022 commented on src/test/java/com/newworld/saegil/user/service/UserServiceTest.java:
```diff
@@ -1,10 +1,7 @@
-import com.newworld.saegil.user.domain.User;
-import com.newworld.saegil.user.repository.UserRepository;
-import jakarta.persistence.EntityManager;
+import static org.assertj.core.api.Assertions.*;
```
### C
여기도요!

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -152,10 +154,12 @@ class Context_with_valid_token {
                 final User user = new User("권예진", "http://example.com/profile.jpg", "1234567", OAuth2Type.KAKAO);
                 userRepository.save(user);
-                final Token token = tokenProcessor.generateToken(LocalDateTime.now(), new PrivateClaims(user.getId()).toMap());
+                final Token token = tokenProcessor.generateToken(LocalDateTime.now(),
```
### A
여기 클래스에 있는 코드 모두 원래 개행이 더 가독성이 좋아 보여요

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -322,6 +343,28 @@ class Context_with_invalid_token {
     class Describe_refreshToken {
 
+        @Test
+        void 토큰_갱신에_사용한_refreshToken은_블랙리스트에_등록된다() {
```
### R
이 테스트 위치를 제가 잘못 넣어놨었네요. 근데 여기가 아니라 Context_with_valid_refresh_token 클래스 내부로 이동시켜야 합니다. 여긴 Describe 영역이라서요

kwonyj1022 reviewed (changes requested):
fast api 쪽 완성되면 제가 작업하려고 했는데 대신 작업해주셨군요 👍

> 혹시 강제보다 더 안정적인 방법으로 맞추는 방법을 아시나요?
사전에 푸시한 브랜치와 로컬에서 새롭게 rebase된 사항을 푸시 올리는 것을 어떻게 안정적으로 할 수 있죠?

저는 그냥 원격 브랜치를 로컬브랜치로 merge 하는 편입니다.

---

RCA 룰을 사용해서 리뷰를 작성해봤습니다.
- R (Request Changes) : 적극적으로 반영을 고려해주세요
- C (Comment) : 웬만하면 반영해주세요
- A (Approve) : 반영해도 좋고, 넘어가도 좋습니다. 사소한 의견입니다.

---
전체적으로 intellij의 파일 전체 자동 정렬 기능을 사용하신 것 같은데요, 저는 개인적으로 이 기능을 feature PR에서 사용하는 것은 안 좋아 하는 편입니다. 이유는 다음과 같아요.
1. **불필요한 file changes가 많아집니다.**
  1-1. **리뷰에 부담을 줍니다.** 실제 내용은 단순하더라도, file changes가 많으면 "뭔가 복잡할 것 같다"는 인상을 주게 되고, 리뷰 시간이 길어질까봐 부담을 가지게 되는 것 같아요.
  1-2. **PR의 목적과 무관한 파일까지 변경됩니다.** 지금 지웅님 PR의 작업 단위는 `STT-ChatGPT-TTS 엔드포인트 연결`인데, 자동 정렬로 인해 관련 없는 파일도 변경되면서 커밋 이력이 불명확해질 수 있습니다. (기능 구현을 위해 수정해야 파일에서 정렬을 맞추는 것은 괜찮다고 봅니다) 예를 들어 이 pr을 develop 브랜치에 squash merge하고 나면 이 PR에서 작업한 내용이 develop 브랜치에 `feat: STT-ChatGPT-TTS 엔드포인트 연결 (#26)`이라는 커밋 메시지로 남게 됩니다. 그래서 나중에 OrganizationController의 수정 이력을 보면 `feat: STT-ChatGPT-TTS 엔드포인트 연결 (#26)` 이라는 커밋이 있겠죠? 근데 이 커밋에서 도대체 이 파일을 왜 수정했는지 의문이 생기게 됩니다. 이런 일이 자주 있지는 않겠지만, 커밋메시지가 작업 내용을 명확하게 표현하지 못하는 문제가 있다고 생각합니다.
2. **IDE 설정 차이로 인한 불필요한 변경이 발생합니다.**
예를 들어 import 순서, 공백, 개행 등은 개인 설정에 따라 달라집니다. 지웅님의 PR을 merge한 후 제가 제 컴퓨터로 또 자동정렬을 수행하면 import 순서가 원래대로 돌아갈 겁니다. 이걸 그대로 PR을 날리면 지금처럼 또 엄청난 양의 file changes가 생기면서 불필요한 변경 사항이 자꾸 커밋에 포함되게 됩니다. 앞서 말했듯이 수정 범위에 있는 파일이라면 상관 없지만 PR에서 다루지 않는 범위의 파일이 계속 수정되면서 커밋에 포함되는 것은 좋지 않다고 생각해요.
3. **IDE가 개행하는 것이 때로는 더 가독성이 안 좋을 때도 있습니다.** 
저번에 저희가 IDE의 가이드라인 선을 넘지 않도록 하자고 했었는데요, 이건 단순히 가이드라인이기 때문에 한 두자 정도 넘어가는 것은 괜찮다고 봅니다. 특히 테스트코드나 stream에서 메서드 체이닝 시작점이 밀리는 바람에 생기는 약간의 오차는 괜찮다고 봅니다. 그런데 ide는 이런 값을 로봇처럼 개행시켜버려요. 알잘딱깔센이 안 되는 친구입니다.

이런 이유로 저는 전체 파일 자동 정렬은 지양하는 편입니다. 한다면 별도 PR에서 정리하는 것이 좋다고 생각해요. 그리고 혹시나 해서 말씀드리지만 이 이야기는 단순히 제 관점을 공유하기 위한 것이지 원상복구를 요구하는 것은 절대 아닙니다. 그래서 관련한 코멘트는 모두 R, C, A 중 A로 달아두었습니다.

jayn2u commented:
정성스러운 생각 너무 잘 들었습니다. 피처 PR에서 코드 포맷팅 관련한 내용을 아예 담지 않는 것을 생각하실 줄 몰랐어요. 저는 이렇게 PR이 올라오면 해당 커밋을 제외하고, 핵심적인 커밋만 골라서 보면서 리뷰를 했었습니다. 그런데 전부 꼼꼼히 읽어주실 줄 몰랐네요!

해당 부분은 반영해서 조치하겠습니다. 저는 오히려 자동화를 선호하는 편 입니다. 사람이 작업에서 정형화된 규칙을 적용할 때 매뉴얼을 토대로 진행하는데(여기서는 코드 컨벤션을 지켰는지 확인하는 것이겠죠?), 매번 이러한 매뉴얼을 지키고 있는지 사람이 검수하기에는 비용이 많이 드는 작업이라고 생각해요. 그래서 비용을 줄일 수 있는 작업으로 자동화를 선호합니다. 하지만 예진님께서 달아주신 리뷰 중에서 불필요한 개행이라던가, 오히려 더 별로인 조치가 이루어지는 사항들에 대해서는 저도 동의하진 않아요! 그래서 꼼꼼이 정의한 코드 포맷팅 규칙을 담은 파일들은 회사에서 공유하는 듯 합니다. 

[Google Java Style Guide](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml) 나 [Naver Hackday naver-intellij-formatter.xml](https://github.com/naver/hackday-conventions-java/blob/master/rule-config/naver-intellij-formatter.xml)를 본 적이 있거든요!

jayn2u committed:
refactor: 코드 정리 및 불필요한 import 제거
- 테스트 코드 및 주요 클래스에서 불필요한 import 문 정리
- 코드 포맷팅 변경 및 가독성 개선
- 사용되지 않는 LlmErrorResponse 클래스 제거

jayn2u commented on src/main/java/com/newworld/saegil/authentication/controller/AuthenticationController.java:
```diff
@@ -16,13 +16,7 @@
-import org.springframework.web.bind.annotation.RequestHeader;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
와일드카드 확인했습니다. 수정해서 올릴게요!

jayn2u replied:
#27 PR에서 와일드카드를 사용하지 않도록 전면 수정했습니다.

jayn2u reviewed (commented):
전 프로젝트에 걸쳐 와일드카드를 쓰지 않도록 개선해서 PR을 올리겠습니다.

jayn2u commented:
저도 작업하면서 알았는데, Naver-Hackday 코드 스타일이 IDE에서 사용되고 있었더라구요? 그래서 이것을 InteliJ 기본으로 변경했습니다.
그리고 기본에서 복제를 떠서 저희 프로젝트 전용 코드 스타일 설정 파일을 생성했습니다. #27 에서 사용된 부분도 전부 그렇게 진행했습니다.

jayn2u commented:
만약 승인하시면 그냥 병합해주세요!

kwonyj1022 commented on src/main/java/com/newworld/saegil/notice/service/NoticeService.java:
```diff
@@ -57,10 +57,12 @@ private List<Notice> crawlAndExtractNewNotices(final NoticeCrawler crawler, fina
-                                            .filter(newNotice -> latestNotices.stream()
-                                                                              .noneMatch(latestNotice -> latestNotice.hasSameTitle(newNotice))
-                                            ).toList();
+        final List<Notice> newNotices =
+                crawledNotices.stream()
+                        .filter(newNotice ->
+                                latestNotices.stream()
+                                        .noneMatch(latestNotice -> latestNotice.hasSameTitle(newNotice))
+                        ).toList();
```
.의 위치를 맞추는 것이 어디서 체이닝이 걸리는지 명확해서 더 좋다고 생각하는데 어떻게 생각하시나요?

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/domain/OAuth2HandlerCompositeTest.java:
```diff
@@ -1,16 +1,12 @@
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
```
import가 바뀌었어요

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/domain/OAuth2TypeTest.java:
```diff
@@ -7,7 +7,7 @@
 import org.junit.jupiter.api.Test;
 
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
```
여기도 바뀌었네요

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -249,8 +247,10 @@ class Context_with_valid_token {
                 SoftAssertions.assertSoftly(softAssertions -> {
-                    softAssertions.assertThat(authenticationService.isValidToken(TokenType.ACCESS, token.accessToken())).isFalse();
-                    softAssertions.assertThat(authenticationService.isValidToken(TokenType.REFRESH, token.refreshToken())).isFalse();
+                    softAssertions.assertThat(authenticationService.isValidToken(TokenType.ACCESS, token.accessToken()))
+                            .isFalse();
+                    softAssertions.assertThat(
+                            authenticationService.isValidToken(TokenType.REFRESH, token.refreshToken())).isFalse();
```
이 둘이 같은 문장(같은 의미의 테스트)이라 스타일이 통일되는게 좋을 것 같아요

kwonyj1022 commented on src/test/java/com/newworld/saegil/user/domain/UserTest.java:
```diff
@@ -1,14 +1,10 @@
 
-import static org.assertj.core.api.Assertions.assertThatCode;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
```
여기도 바뀌었네요

kwonyj1022 commented on src/test/java/com/newworld/saegil/user/service/UserServiceTest.java:
```diff
@@ -5,17 +5,14 @@
 import org.springframework.transaction.annotation.Transactional;
 
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
```
여기도 바뀌었네요

kwonyj1022 commented on src/test/java/com/newworld/saegil/user/domain/UserTest.java:
```diff
@@ -30,7 +26,7 @@ class Context_with_valid_input {
             @Test
             void 예외가_발생하지_않는다() {
-                assertThatCode(() -> new User(validName, validProfileImageUrl, validOAuth2Id, validOAuth2Type))
+                Assertions.assertThatCode(() -> new User(validName, validProfileImageUrl, validOAuth2Id, validOAuth2Type))
```
여기는 왜 static import를 안 하는 것으로 바꾸었나요?

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -188,21 +176,25 @@ class Context_with_invalid_token {
                 blacklistTokenRepository.save(new BlacklistToken(user.getId(), TokenType.ACCESS, token.accessToken()));
-                blacklistTokenRepository.save(new BlacklistToken(user.getId(), TokenType.REFRESH, token.refreshToken()));
+                blacklistTokenRepository.save(
+                        new BlacklistToken(user.getId(), TokenType.REFRESH, token.refreshToken()));
```
given 절은 비즈니스로직이 아니라 단순한 "데이터 세팅"이기 때문에 "save"의 의미가 더 중요하고, 한 줄에 읽히는 게 중요하다고 생각해서 일부러 개행을 안 한 거긴 합니다. 그래도 만약 개행을 해야 한다면 아래처럼 하는게 더 좋을 것 같아요.
파라미터를 개행하는 경우 괄호의 시작과 끝, 각 파라미터 값의 시작과 끝이 명확한 걸 더 선호합니다.

```suggestion
                blacklistTokenRepository.save(
                        new BlacklistToken(user.getId(), TokenType.REFRESH, token.refreshToken())
                );
```

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -188,21 +176,25 @@ class Context_with_invalid_token {
                 userRepository.save(user);
-                final Token token = tokenProcessor.generateToken(LocalDateTime.now(), new PrivateClaims(user.getId()).toMap());
+                final Token token = tokenProcessor.generateToken(LocalDateTime.now(),
+                        new PrivateClaims(user.getId()).toMap());
```
이 부분도 마찬가지 입니다. 그래서 개행을 한다면 아래처럼 하는 것을 더 선호합니다.
```suggestion
                final Token token = tokenProcessor.generateToken(
                        LocalDateTime.now(),
                        new PrivateClaims(user.getId()).toMap()
                );
```

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/service/AuthenticationServiceTest.java:
```diff
@@ -188,21 +176,25 @@ class Context_with_invalid_token {
-                    softAssertions.assertThatThrownBy(() -> authenticationService.getValidPrivateClaims(TokenType.REFRESH, token.refreshToken()))
-                                  .isInstanceOf(InvalidTokenException.class)
-                                  .hasMessage("사용할 수 없는 토큰입니다.");
+                    softAssertions.assertThatThrownBy(
+                                    () -> authenticationService.getValidPrivateClaims(TokenType.ACCESS, token.accessToken()))
+                            .isInstanceOf(InvalidTokenException.class)
+                            .hasMessage("사용할 수 없는 토큰입니다.");
```
여기도 마찬가지입니다. assertThatThrownBy, isInstanceOf, hasMessage 가 중요하기 때문에 개행을 해서 중간에 흐름을 끊기 보다는 isInstanceOf가 assertThatThrownBy 바로 다음 줄에 위치하는게 더 가독성이 좋다고 판단했습니다. 이렇게 해도 assertThatThrownBy 절이 지나치게 길어지지 않으니까요.

그래도 개행을 해야 한다면 아래처럼 assertThatThrownBy절 괄호의 시작과 끝을 명확히 해주는게 좋다고 생각합니다. 지금 코드로 읽다보면 괄호를 하나하나 세지 않는 이상 .isInstanceOf가 .getValidPrivateClaims() 뒤에 걸리는 것인지 assertThatThrownBy() 뒤에 걸리는 것인지 명확하지 않다고 생각해요.
```suggestion
                    softAssertions.assertThatThrownBy(
                                          () -> authenticationService.getValidPrivateClaims(TokenType.ACCESS, token.accessToken())
                                  ).isInstanceOf(InvalidTokenException.class)
                                  .hasMessage("사용할 수 없는 토큰입니다.");
```

kwonyj1022 commented on src/test/java/com/newworld/saegil/authentication/domain/OAuth2HandlerCompositeTest.java:
```diff
@@ -47,7 +43,7 @@ class Context_invalid_oauth2_type {
                 final OAuth2HandlerComposite emptyComposite = new OAuth2HandlerComposite(Set.of());
 
-                // when & then
+                // when and then
```
`// when and then`으로 바꾸는 것은 상관 없지만 바꾼다면 통일성 있게 다른 쪽도 수정하면 좋을 것 같아요. 지금은 여기만 바뀌어 있네요

kwonyj1022 reviewed (changes requested):
> 기본에서 복제를 떠서 저희 프로젝트 전용 코드 스타일 설정 파일을 생성했습니다.

이건 어디서 확인할 수 있나요? 저도 같이 적용한다면 import 순서가 계속 바뀌는 문제는 해결할 수 있겠네요. 지금은 제가 이 브랜치에서 자동 정렬하면 import 순서가 다시 원래대로 돌아가서요.

\+ 어떻게 설정이 되어있는지 모르겠지만, 코드가 가로로 길어지는 경우에 대한 개행 관련해서는 설정을 제거하거나, commit 전에 개발자가 검수를 하는게 좋을 것 같다고 생각해요. 저는 개행도 의도를 가진다고 생각해요. 그래서 개행을 안 하는게 더 좋은 경우, 인텔리제이의 가이드라인이 예외가 되는 경우가 존재한다고 생각합니다. 이 부분은 pr 단위에서 합의해야할 사항이지 자동화에 맡길 수 있는 부분은 아니라고 생각해요.

jayn2u commented:
> > 기본에서 복제를 떠서 저희 프로젝트 전용 코드 스타일 설정 파일을 생성했습니다.
> 
> 이건 어디서 확인할 수 있나요? 저도 같이 적용한다면 import 순서가 계속 바뀌는 문제는 해결할 수 있겠네요. 지금은 제가 이 브랜치에서 자동 정렬하면 import 순서가 다시 원래대로 돌아가서요.
> 
> + 어떻게 설정이 되어있는지 모르겠지만, 코드가 가로로 길어지는 경우에 대한 개행 관련해서는 설정을 제거하거나, commit 전에 개발자가 검수를 하는게 좋을 것 같다고 생각해요. 저는 개행도 의도를 가진다고 생각해요. 그래서 개행을 안 하는게 더 좋은 경우, 인텔리제이의 가이드라인이 예외가 되는 경우가 존재한다고 생각합니다. 이 부분은 pr 단위에서 합의해야할 사항이지 자동화에 맡길 수 있는 부분은 아니라고 생각해요.

<img width="1169" alt="Screenshot 2025-04-20 at 22 34 42" src="https://github.com/user-attachments/assets/b54ce57b-074a-4fc4-9f36-b7731e42e2ca" />

코드 스타일 파일은 별도의 `xml`로 파일로 공유하면 돼요! 프로젝트 자체에는 삽입할 수 없는 듯 합니다...
